### DISCUSSION TITLE
Is there a way to close the terminal after launching a GUI process from inside lf?

### DISCUSSION THREAD

**Discussion by Archie-2021** [gokcehan/lf#1487] [Q&A] [answered]
I am trying to automatically close the terminal after opening a pdf file to get it out of the way. I can fork the process so that new window don't get closed after lf process ends , but I want to also close lf and the terminal after opening the terminal. I can't find a solution. running something like `kill $$` works in terminal but not from inside lf.

**Comment by gokcehan** [comment]
@Archie-2021 I haven't tested it, but does something like the following work?

```
map L :open; quit
```

**Reply by joelim-work** [reply]
If you want to close the terminal after quitting `lf`, you can start the terminal to just run `lf` directly instead of the default shell. For example with [alacritty](https://github.com/alacritty/alacritty), use `alacritty -e lf`.

On a side note, I'm not sure if we have a section in the wiki about using `lf` as a file picker/opener. The following works for me:

```shell
alacritty -e sh -c 'setsid xdg-open "$(lf -print-selection)"'
```

Hopefully it works for others, or at the very least serves as a useful starting point.

**Reply by Archie-2021** [reply]
Hey, and thanks for you help

Yes if i launch lf via `alacritty -e lf` it would close it when i quite lf. but what I qm trying to do is to open the file under the cursor with a certain program and then quite (I have to also fork the process when opening the pdf file so that the pdf stays open after terminal and lf closed. But from what I have tried even when I close lf in this way the terminal stays open. That is why I am asking.
Also can you the link to wiki section you talked about . I am actually thing about how to force my browser to open newly downloaded files in lf instead of default GUI


Thanks

**Reply by joelim-work** [reply]
Oh, if you want to open your terminal normally with a shell, run `lf` and then quit the whole terminal on-demand, then I'm not sure if there's a simple way to do this. Normally, programs don't try to kill the terminal that they run in. I think you might have to invoke your window manager to locate and kill the window, or use `ps` to find the PID of the terminal and kill it.

**Reply by joelim-work** [reply, marked as answer]
I forgot to mention this earlier, but from the way you describe your workflow, it sounds like you might be interested in window swallowing, for example using [`devour`](https://github.com/salman-abedin/devour).

> Devour hides your current window before launching an external program and unhides it after quitting.

I haven't used this before, but I can see the appeal behind it.

**Reply by Archie-2021** [reply]
> I forgot to mention this earlier, but from the way you describe your workflow, it sounds like you might be interested in window swallowing, for example using [`devour`](https://github.com/salman-abedin/devour).

Thank you. That is exactly what I was looking for. such a nice solution to this problem

**Comment by Archie-2021** [comment]
> I forgot to mention this earlier, but from the way you describe your workflow, it sounds like you might be interested in window swallowing, for example using [`devour`](https://github.com/salman-abedin/devour).
> 
> > Devour hides your current window before launching an external program and unhides it after quitting.
> 
> I haven't used this before, but I can see the appeal behind it.
